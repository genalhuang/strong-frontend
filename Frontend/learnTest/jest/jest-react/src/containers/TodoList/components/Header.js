import React, {Component} from 'react'


class Header extends Component {
  constructor(props) {
    super(props)
    this.state = {
      value: ''
    }
    this.handleInputChange = this.handleInputChange.bind(this)
    this.handleInputKeyUp = this.handleInputKeyUp.bind(this)
  }

  handleInputChange(e) {
    this.setState({
      value: e.target.value
    })
  }

  handleInputKeyUp(e) {
    const value = this.state.value
    if(e.keyCode === 13 && value) {
      this.props.addUndoItem(value);
    }
  }

  render() {
    const {value} = this.state;

    return (
      <div>
        <input  data-test='input' value={value} onChange={this.handleInputChange} onKeyUp={this.handleInputKeyUp}></input>
      </div>
    )
  }
}

export default Header;
