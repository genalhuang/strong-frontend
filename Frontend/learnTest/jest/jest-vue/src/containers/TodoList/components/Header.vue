<template>
  <div class="header">
    <input data-test='input' v-model='inputValue' @keyup.enter='addTodoItem'>
  </div>
</template>

<script>
export default {
  name: 'Header',
  data() {
    return {
      inputValue: ''
    }
  },
  methods: {
    addTodoItem() {
      if(this.inputValue) {
        this.$emit('add', this.inputValue)
        this.inputValue = '';
      }
    }
  }
}
</script>

<!-- Add "scoped" attribute to limit CSS to this component only -->
<style scoped lang="stylus">
.header {
  background-color: blue;
}
</style>
