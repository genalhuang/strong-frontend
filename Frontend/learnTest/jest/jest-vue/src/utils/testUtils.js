export const findTestWrapper = (wrapper, tag) => {
  return wrapper.findAll(`[data-test="${tag}"]`)
}