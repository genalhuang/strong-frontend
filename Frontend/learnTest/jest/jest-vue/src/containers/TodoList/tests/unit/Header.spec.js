import Header from '../../components/Header.vue'
import { shallowMount } from '@vue/test-utils'
import {findTestWrapper} from '@/utils/testUtils'

it('Header 样式改变做提示', () => {
  const wrapper = shallowMount(Header)
  expect(wrapper).toMatchSnapshot();
})

it('Header 包含 input 框', () => {
  const wrapper = shallowMount(Header)
  const input = findTestWrapper(wrapper,'input');
  expect(input.exists()).toBe(true);
})

it('Header 中 input 框初始内容为空', () => {
  const wrapper = shallowMount(Header)
  const inputValue = wrapper.vm.$data.inputValue;
  expect(inputValue).toBe('');
})

it('Header 中 input 框值发生变化，数据应该跟着变', () => {
  const wrapper = shallowMount(Header)
  const input = findTestWrapper(wrapper,'input');
  input.setValue('genal')
  const inputValue = wrapper.vm.$data.inputValue;
  expect(inputValue).toBe('genal');
})

it('Header 中 input 框输入回车无内容时无反应', async () => {
  const wrapper = shallowMount(Header)
  const input = findTestWrapper(wrapper,'input');
  await input.setValue('')
  input.trigger('keyup.enter');
  expect(wrapper.emitted().add).toBeFalsy();
})

it('Header 中 input 框 如果有内容时向外触发事件', async () => {
  const wrapper = shallowMount(Header)
  const input = findTestWrapper(wrapper,'input');
  await input.setValue('1234')
  input.trigger('keyup.enter');
  expect(wrapper.emitted().add).toBeTruthy();
})


it('Header 中 input 框 如果有内容时向外触发事件, 同时清空inputValue', async () => {
  const wrapper = shallowMount(Header)
  const input = findTestWrapper(wrapper,'input');
  await input.setValue('1234')
  input.trigger('keyup.enter');
  expect(wrapper.emitted().add).toBeTruthy();
  expect(wrapper.vm.$data.inputValue).toBe('')
})