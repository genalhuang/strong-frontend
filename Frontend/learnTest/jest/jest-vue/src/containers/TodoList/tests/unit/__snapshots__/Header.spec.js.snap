// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header 样式改变做提示 1`] = `<div class="header"><input data-test="input"></div>`;
